core/src/banking_stage.rs

```diff
@@ -433,8 +433,8 @@ impl BankingStage {
         if unprocessed_transaction_storage.should_not_process() {
             return;
         }
-        let ((metrics_action, decision), make_decision_time) =
-            measure!(decision_maker.make_consume_or_forward_decision(slot_metrics_tracker));
+        let (decision, make_decision_time) =
+            measure!(decision_maker.make_consume_or_forward_decision());
```

Comment:
hmm, `.check_leader_slot_boundary()` won't be called for the case of `BufferedPacketsDecision::Hold`. So, technically this is breaking change with regard to the precise definition of those metrics. (as i ranted earlier at banking trace pr, i wonder this is worth to retain so insistently....)

Anyway, I'd suggest to call `.check_leader_slot_boundary()` right here mainly for simplicity and maintenance ease:

```rust
let (decision, make_decision_time) =
    measure!(decision_maker.make_consume_or_forward_decision());
let metrics_action = slot_metrics_tracker.check_leader_slot_boundary(decision.bank_start());


...


impl BufferedPacketDecision {
  fn bank_start(&self) -> &Option<BankStart> {
       match self { ... }
  }
}

```

Reply:
I think it's likely better to keep the type juggling change - `bank_start(&self) -> &Option<BankStart>` would require cloning the `BankStart` (2 `Arc`s). But we can just copy the reference into an option, and avoid those clones 😄